Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py

```diff
@@ -2071,10 +2073,24 @@ def handle_attached_email_with_incorrect_id(attached_email: Message):
             break
     if message_id_value:
         # If the Message-ID header was fixed in the context of this function, it will be inserted again to the _headers list
+        demisto.debug(f'XSUP-32660: {message_id_value=}')
         attached_email._headers.append(("Message-ID", message_id_value))
     return attached_email
 
 
+def handle_incorrect_message_id(message_id):
+    """
+    Handles the same logic as handle_attached_email_with_incorrect_message_id but expects only a string.
+    """
+    demisto.debug('XSUP-32660: handle_incorrect_message_id running')
+    if message_id.find("<[") != -1 or message_id.find("]>") != -1:
```

Comment:
instead of repeating the find logic, consider a method

Reply:
Done.
`handle_attached_email_with_incorrect_message_id` function is now using the `handle_incorrect_message_id` logic, thus removing the redundant use of duplicate code.